napari/layers/_shapes_layer/model.py

```diff
@@ -0,0 +1,1568 @@
+
+        return rescale.mean()
+
+    def _get_coord(self, position):
+        """Convert a position in canvas coordinates to image coordinates.
+
+        Parameters
+        ----------
+        position : sequence of int
+            Position of mouse cursor in canvas coordinates.
+
+        Returns
+        ----------
+        coord : sequence of float
+            Position of mouse cursor in image coordinates.
+        """
+        transform = self.viewer._canvas.scene.node_transform(self._node)
+        pos = transform.map(position)
+        coord = np.array([pos[0], pos[1]])
+        self._cursor_coord = coord
+
+        return coord
+
+    def get_message(self, coord, shape, vertex):
+        """Generates a string based on the coordinates and information about
+        what shapes are hovered over
+
+        Parameters
+        ----------
+        coord : sequence of int
+            Position of mouse cursor in image coordinates.
+        shape : int | None
+            Index of shape if any to be highlighted.
+        vertex : int | None
+            Index of vertex if any to be highlighted.
+        Returns
+        ----------
+        msg : string
+            String containing a message that can be used as a status update.
+        """
+        coord_shift = copy(coord)
+        coord_shift[0] = int(coord[1])
+        coord_shift[1] = int(coord[0])
+        msg = f'{coord_shift.astype(int)}, {self.name}'
+        if shape is not None:
+            msg = msg + ', shape ' + str(shape)
+            if vertex is not None:
+                msg = msg + ', vertex ' + str(vertex)
+        return msg
+
+    def move_to_front(self):
+        """Moves selected objects to be displayed in front of all others.
+        """
+        if len(self.selected_shapes) == 0:
+            return
+        new_z_index = max(self.data._z_index) + 1
+        for index in self.selected_shapes:
+            self.data.update_z_index(index, new_z_index)
+        self.refresh()
+
+    def move_to_back(self):
+        """Moves selected objects to be displayed behind all others.
+        """
+        if len(self.selected_shapes) == 0:
+            return
+        new_z_index = min(self.data._z_index) - 1
+        for index in self.selected_shapes:
+            self.data.update_z_index(index, new_z_index)
+        self.refresh()
+
+    def _copy_shapes(self):
+        """Copy selected shapes to clipboard.
+        """
+        self._clipboard = ([deepcopy(self.data.shapes[i]) for i in
+                           self._selected_shapes])
+
+    def _paste_shapes(self):
+        """Paste any shapes from clipboard and then selects them.
+        """
+        cur_shapes = len(self.data.shapes)
+        for s in self._clipboard:
+            self.data.add(s)
+        self.selected_shapes = list(range(cur_shapes,
+                                    cur_shapes+len(self._clipboard)))
+        self.move_to_front()
+        self._copy_shapes()
+
+    def _move(self, coord):
```

Comment:
Overall comment: Breaking up some of these huge methods into smaller helper ones would make things easier to read and understand.

Reply:
it's a little unclear to me how to break this up in a way that makes sense

Reply:
one way to do it would be to abstract all the if/else statements into their own methods that describe what each clause is doing something like 
```
def move():
    if self.mode in ([Mode.SELECT, Mode.ADD_RECTANGLE, Mode.ADD_ELLIPSE, Mode.ADD_LINE]):
        if vertex is None:
            move_whole_onject()
        if vertex < BOX_LEN:
            resize_object()
        if vertex == 8:
            resize_object()
    elif self.mode in [Mode.DIRECT, Mode.ADD_PATH, Mode.ADD_POLYGON]:
        if len(self.selected_shapes) > 0:
            # not exactly sure how to describe this part
        else: 
            # same?
    elif self.mode in [Mode.VERTEX_INSERT, Mode.VERTEX_REMOVE]:
        add_or_remove_vertex()
```
I'd be down to pair with you on this if you want